nanoFramework.System.Net.Http/Http/System.Net.HttpListener.cs

```diff
@@ -461,17 +461,38 @@ private void AcceptThreadFunc()
         /// the <see cref='Stop'/> method to stop it.
         /// </remarks>
         public void Start()
+        {
+            IPAddress addr = IPAddress.GetDefaultLocalAddress();
+            Start(addr.ToString());
+        }
+
+        /// <summary>
+        /// Allows this instance to receive incoming requests.
+        /// </summary>
+        /// <param name="niIpAddress">
+        /// IP address of the intended network interface.
+        /// If there are several available network interfaces, such as Ethernet, SoftAp, and WiFi, 
+        /// you can target a specific one.
+        /// </param>
+        /// <remarks>This method must be called before you call the
+        /// <see cref="GetContext"/> method.   If
+        /// the service was already started, the call has no effect.  After you
+        /// have started an <itemref>HttpListener</itemref> object, you can use
+        /// the <see cref='Stop'/> method to stop it.
+        /// </remarks>
+        /// <summary>
+        public void Start(string niIpAddress)
```

Comment:
while you are there, why not then directly use an IPAddress class rather then the string?

Reply:
That can also be done. It is a matter of choice.
Basically my motivation was that the string is simpler than the class:)

Reply:
Like I am getting the SoftAP IP string form a config JSON file and call Start directly with the string.

Reply:
Can you then also add a constructor with the IPAddress? It makes things a bit cleaner when it comes from getting it from the system for example. So it will avoid a conversion to string and then a parsing.

Reply:
How about this then:
add:    private readonly IPAddress _addr;

add:        public HttpListener(string prefix, IPAddress addr, int port)
           _addr = addr;

           /// Other ctors will    _addr = IPAddress.GetDefaultLocalAddress
           
revert Start() back to original and replace addr with _addr
?